driver/xrt/src/accl.cpp

```diff
@@ -460,6 +479,8 @@ CCLO *ACCL::scatter(BaseBuffer &sendbuf,
                     communicatorId comm_id, bool from_fpga, bool to_fpga,
                     dataType compress_dtype, bool run_async,
                     std::vector<CCLO *> waitfor) {
+  check_tcp_ready();
```

Comment:
it's unlikely we need a check for each call right? if we do, wouldn't it be more appropriate to put it in e.g. prepare_call?